fastestimator/op/tensorop/gradient/gradient.py

```diff
@@ -74,9 +77,19 @@ def forward(self, data: List[Tensor], state: Dict[str, Any]) -> List[Tensor]:
                 results.append(get_gradient(final, initial, tape=state['tape'], retain_graph=retain_graph))
         else:
             finals = data
-            trainable_params = [p for p in self.model.parameters() if p.requires_grad] if isinstance(
-                self.model, torch.nn.Module) else self.model.trainable_variables
-            for idx, final in enumerate(finals):
-                retain_graph = self.retain_graph or not idx == len(finals) - 1
-                results.append(get_gradient(final, trainable_params, tape=state['tape'], retain_graph=retain_graph))
+            if self.framework == "tf":
```

Comment:
should we have a get_params backend function so that we can eliminate this if check? in general it's preferable if ops can avoid framework branching since it reduces likelihood of divergent functionality between frameworks

Reply:
I agree with the idea of making the get_params backend function. But I don't think we can fully avoid the framework branching in such framework-sensitive Op.